### DISCUSSION TITLE
use for recurrent workflows

### DISCUSSION THREAD

**Discussion by xinstein** [ropensci/targets#674] [Help]
I'm doing a longitudinal modelling problem (stock prediction), which involves a bunch of targets all involving date as tag, some of which depends on a windowed range of another. as a demonstration:

targ_a_20200825
targ_a_20200826
targ_a_20200827
targ_a_20200828
...

targ_b_20200827 depends on targ_a_20200825 ~ targ_a_20200827
targ_b_20200828 depends on targ_a_20200826 ~ targ_a_20200828
...

I'm tryin to find a proper way of handling this kind of dependency graph (I'll call this a recurrent workflow before I find any officially accepted name for it).

I'm considering using static branching or dynamic branching to achieve this, but before I start, I pondered a little and found myself the following problems and concerns:

1. meta-programming might be painstaking, especially for static branching
2. some of the dependency window lengths is as large as a couple hundreds, and the date tag runs through a few years (thousands of days), I read about this ( https://github.com/ropensci/targets/issues/246 ) and am afraid of unacceptible overhead (both dependency resolution and the job itself).
3. for dynamic branching the target names are unreadible

I have several reasons to believe there's a better way to handle this than simple well documented static/dynamic branching:
1. there's recurrence in the dependency graph, basically targets is rediscovering the same dependency pattern for each day's target, giving rise to a lot of unnecessary overhead.
2. each day's data is stored separately in spite of having the same structure, redoing almost-identical data union operations each day.

I'm hoping for a proper resolution for this kind of problem. Please let me know your thoughts or suggestions, I'm willing to try.

**Comment by wlandau** [comment]
A technique like https://github.com/ropensci/targets/discussions/615#discussioncomment-1239073 could help. Basically, you could build up an adjacency matrix in the `values` list and then plug that into either `tar_map()` or `tar_eval()` from `tarchetypes`. (`tar_eval()` makes it easier to control target names.)

Alternatively, depending on the graph topology, you might be able to batch sequences of analyses into individual targets. For example, you could have one target for targ_a_20200825, targ_a_20200827, and targ_b_20200827, and another target for
targ_b_20200828, targ_a_20200826, and targ_a_20200828. This would make sense if all of targ_a_20200825, targ_a_20200827, and targ_b_20200827 were completely independent from all of targ_b_20200828, targ_a_20200826, and targ_a_20200828. Otherwise, if there are likely to be some cross-dependencies, this would not be practical.
